### DISCUSSION TITLE
Waveform doesnt render whenever inside flex container

### DISCUSSION THREAD

**Discussion by shulinnn** [katspaugh/wavesurfer.js#3211] [Q&A] [answered]
Hello. Whenever i try to put waveform inside flex container it simply does not render anything.
index.html
```
<body class="bg-black flex">
  <div id="player" class="h-screen"></div>
  <script type="module" src="/main.js"></script>
</body>
```

main.js
```
const wavesurfer = w.create({
  container: "#player",
  waveColor: "white",
  progressColor: "gray",
  fillParent: true,
  url: a,
});

wavesurfer.once("interaction", () => {
  wavesurfer.play();
});
```

**Comment by katspaugh** [comment]
You need to set a width on the container. Wavesurfer doesn't have its own width, it's always taken from the parent div.

**Reply by shulinnn** [reply]
Well after setting the width for the container , it still doesnt show anything. If i just remove the `display:flex;` it shows the waveform.

**Reply by katspaugh** [reply, marked as answer]
Check that you're setting the width on the right container. Here's a working example: https://wavesurfer-js.org/examples/?b6e1c5ec474258ff46200064dc57ad21

**Reply by shulinnn** [reply]
That was it. Thanks a lot.
